src/halmos/config.py

```diff
@@ -549,7 +549,15 @@ def parse_dict(self, parsed: dict, source: str = "halmos.toml") -> dict:
                 )
                 sys.exit(2)
 
-        return {k.replace("-", "_"): v for k, v in data.items()}
+        def parse_value(k, v):
+            if k == "default-bytes-lengths":
+                return ParseCSV.parse(v)
+            elif k == "array-lengths":
+                return ParseArrayLengths.parse(v)
+            else:
+                return v
+
+        return {k.replace("-", "_"): parse_value(k, v) for k, v in data.items()}
```

Comment:
@karmacoma-eth this logic requires to be updated whenever a new custom parser is added. is there an efficient way to search the custom parser, possibly avoiding a linear scan of `fields(Config)`?